tools/github/write_current_source_version.py

```diff
@@ -54,9 +66,14 @@
         sys.exit(1)
 
     # Fetch the current version commit information from GitHub.
-    with urllib.request.urlopen("https://api.github.com/repos/envoyproxy/envoy/commits/v"
-                                + current_version) as response:
-        commit_info = json.loads(response.read())
-        source_version_file = pathlib.Path("SOURCE_VERSION")
-        # Write the extracted current version commit hash "sha" to SOURCE_VERSION.
-        source_version_file.write_text(commit_info["sha"])
+    commit_info_request = urllib.request.Request(
+        "https://api.github.com/repos/envoyproxy/envoy/commits/v" + current_version)
+    github_token = os.environ.get(args.github_api_token_env_name)
+    if github_token != None:
+        # To avoid rate-limited API calls.
+        commit_info_request.add_header("Authorization", "Bearer " + github_token)
+    response = urllib.request.urlopen(commit_info_request)
```

Comment:
im not so familiar with urllib.request as i usually use a higher level lib - but i think preferably use the context - as you had it before - ie:

```python
with ...urlopen(url) as response:
    use_response(response)
```